### ISSUE TITLE
Warning CS0675 fired in Dev12 but not Dev14

### ISSUE THREAD

**Issue by ellismg** [dotnet/roslyn#909]
The following code generates CS0675 when built with Dev12, but doesn't on Dev14 (I think I am using the Dev14 CPT 5 build.  The build number is 14.0.22512.0 DP).

```csharp

long bits = 0;
for (int i = 0; i < 32; i++)
{
    if (i % 2 == 0)
    {
        bits |= (1 << i);
    }
}
```

I'm not sure if this is a regression or by design change.

**Comment by gafter** [comment]
Labeling "Compatibility" as it would be hard to add this warning back after we release a compiler that does not detect it.

**Comment by theoy** [comment]
Approved for 1.0-stable.

**Comment by AlekseyTs** [comment]
@VSadov Please verify
